File: go/fibonacci.go
package main

import (
    "fmt"
    "time"
)

func timeIt(f func()) {
    start := time.Now()
    f()
    elapsed := time.Since(start)
    fmt.Println("Elapsed:", elapsed)
}

func fib(n int) int {
    if (n < 2) {
        return 1
    }
    return fib(n - 1) + fib(n - 2)
}

func main() {
    timeIt(func() { fib(40) })
}

File: c/fibonacci.c
#include <stdio.h>
#include <sys/time.h>

static int fib(int n)
{
    if (n < 2)
        return 1;

    return fib(n - 2) + fib(n - 1);
}

int main(int argc, char *argv[])
{
    struct timeval start, end;
    double elapsed;
    int answer;

    (void) gettimeofday(&start, NULL);
    answer = fib(42);
    (void) gettimeofday(&end, NULL);

    elapsed = (end.tv_sec   * 1000 + end.tv_usec   / 1000.0) -
              (start.tv_sec * 1000 + start.tv_usec / 1000.0);

    printf("%d, elapsed: %.3f msec\n", answer, elapsed);
    return 0;
}
